includes/class-shortcodes.php

```diff
@@ -448,14 +451,41 @@ public function sc_listings_featured( $atts ) {
             'publish',
             WPBDP_POST_TYPE
         );
-        $featured = $wpdb->get_col( $q );
+		$featured   = $wpdb->get_col( $q );
+		$query_args = array(
+			'post__in' => ! empty( $featured ) ? $featured : array( 0 ),
+			'orderby'  => 'post__in',
+		);
+		if ( $sc_atts['category'] ) {
```

Comment:
Can this be moved into a new function to make this ore readable?

Reply:
Introduced a new function `process_category_atts` that handles the category attributes. Using it in `sc_listings_featured` and `sc_listings` function